docs/reference/xpkg.md

```diff
@@ -0,0 +1,201 @@
+sections: requirements related to OCI image format and requirements related to
+Crossplane `package.yaml` contents.
+
+- [OCI Image Format](#oci-image-format)
+  - [Indexes](#indexes)
+  - [Manifests](#manifests)
+  - [Configuration](#configuration)
+  - [Layers](#layers)
+- [package.yaml Contents](#packageyaml-contents)
+  - [Configuration Package Requirements](#configuration-package-requirements)
+  - [Provider Package Requirements](#provider-package-requirements)
+  - [Object Annotations](#object-annotations)
+
+## OCI Image Format
+
+OCI images are comprised of [manifests], [configuration], and [layers].
+Additionally, an image reference could refer to an image [index], which may
+reference multiple image manifests and is frequently used for multi-platform
+images. A valid Crossplane `xpkg` imposes various requirements on the components
+of an OCI, each of which are described in the following sections.
+
+### Indexes
+
+The components of an `xpkg` that Crossplane interacts with do not contain any
+platform-specific information, so Crossplane is broadly agnostic to the
+formatting of an image index. Crossplane does impose the following requirements
+on an image index:
+
+- At least one (1) manifest MUST be referenced in the manifest descriptor array
+  for a package to be successfully fetched and processed.
+
+> The OCI image specification allows for zero-length manifest descriptor arrays
+> in an index.
+
+The following default behavior when interacting with image indexes is
+implemented in the Crossplane package manager:
+
+- If one manifest is referenced in the image index, the image it points to will
+  be used.
+- If multiple manifests are referenced in the image index, Crossplane will use
+  the `linux/amd64` variant by default.
+
+> It is important to note that the platform of the package image that is used by
+> Crossplane does not necessarily mean that the same platform will be used for
+> the controller if the package is a Provider. The decision of selecting a
+> platform for a Provider controller image is deferred to the configured
+> container runtime.
+
+### Manifests
+
+A manifest defines the layers and configuration of a specific image. Crossplane
+is only concerned with the layer descriptors array in an image manifest and does
+not impose additional requirements on any other portions of the manifest. The
+following requirements are imposed on the layer descriptors array:
+
+- One (1) layer descriptor in the array MAY have an [annotation] with key
+  `io.crossplane.xpkg` and value `base`.
+- Any number of layer descriptors in the array MAY have an annotation with key
+  `io.crossplane.xpkg` and arbitrary value. Whether multiple layer descriptors
+  may have the same value is left to the specification of the consumer of those
+  layers.
+
+> As evidenced by the fact that annotations are provided as a map of
+> _string-string_, no single descriptor will contain multiple
+> `io.crossplane.xpkg` annotations.
```

Comment:
Is something like `io.crossplane.xpkg/type` allowed as a key? If so, would that level of specificity make sense to allow for other xpkg related annotations?

Reply:
@negz my reading of [the spec](https://github.com/opencontainers/image-spec/blob/main/annotations.md#rules) is no -- the key must be a "reverse domain name". The value on the other hand is free form. In practice I doubt any tooling rejects annotation keys of a different form (i.e. I would be that they are treated as opaque strings), but I would not want to deviate from the spec due to the fact that Crossplane package images could potentially be deemed non-conforming by tooling in the future.